packages/gatsby-dev-cli/src/watch.js

```diff
@@ -0,0 +1,51 @@
+const chokidar = require(`chokidar`)
+const _ = require(`lodash`)
+const fs = require(`fs-extra`)
+const syspath = require(`path`)
+
+const ignoreRegs = [/[\/\\]node_modules[\/\\]/i, /\.git/i, /[\/\\]src[\/\\]/i]
```

Comment:
while we are it there is a potential bug here that I hit. this will test for `src` in any part of the path, all my local repos are cloned to `~/src/` so the dev-cli never watches any files. Can we change this to be defined in the `ignored` function and do something like `${p}\/src`